frontend/src/RecordingPageComp/RecordingComp.jsx

```diff
@@ -42,11 +64,12 @@ function RecordingComp() {
         audioContextRef.current.decodeAudioData(arrayBuffer, (buffer) => {
           const input = buffer.getChannelData(0); // mono audio
           const detector = PitchDetector.forFloat32Array(input.length);
-          const [pitch, clarity] = detector.findPitch(
+          const [pitch] = detector.findPitch(
             input,
             audioContextRef.current.sampleRate
           );
-          console.log(`Detected pitch: ${pitch} Hz, Clarity: ${clarity}`);
+          frequencyToMIDINoteNumber(pitch);
```

Comment:
Based on the name of the function, I would think that this function just handles conversion of pitch to MIDI note, but it actually sets the note state variable as well. I'd recommend using this function to only handle conversion and setting the note to the returned value outside of the function OR make sure to name the function such that it gives an indication of all of its functionality